Research/lva-node-app/app.js

```diff
@@ -0,0 +1,331 @@
+    });    
+  }).catch((error) =>
+  {
+    console.error(error);
+    res.send("Error: "+error);
+  })
+}) 
+
+
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// defined functions
+
+/**
+* set device ID and IoTHub connection string
+* try to connect to the iot hub to make sure that the passed in values are valid!
+* @returns {Promise<any>} - Promise to resolve/reject based on validity of Device ID and Iot Hub Connection String
+*/
+function setConfigs(req, res)
+{
+  return new Promise((resolve, reject) =>
+  {
+    DEVICE_ID=req.body["device-id"];
+    IOTHUB_CONNECTION_STRING=req.body["iothub-connection-string"];
+    if (DEVICE_ID == "" || IOTHUB_CONNECTION_STRING == "")
+    {
+      DEVICE_ID="";
+      IOTHUB_CONNECTION_STRING="";
+      reject("You must provide both your device ID and your Iot Hub Connection String");
+    }
+    convertIotHubToEventHubsConnectionString(IOTHUB_CONNECTION_STRING).then((eventhubString) =>
+    {
+      IOTHUB_ENDPOINT = eventhubString;
+      resolve();
+    }).catch((error) =>
+    {
+      //reset values on failure
+      console.error(error);
+      DEVICE_ID="";
+      IOTHUB_CONNECTION_STRING="";
+      reject(error.message);
+    });
+  })
+}
+
+
+/** 
+* broadcast used for sending IoT Hub message data in a live stream 
+* @param {any} data - data to send through websocket
+*/
+wss.broadcast = (data) => 
+{
+    wss.clients.forEach((client) => 
+    {
+      if (client.readyState === WebSocket.OPEN) 
+      {
+        try 
+        {
+          client.send(data);
+        } catch (error) 
+        {
+          console.error(error);
+        }
+      }
+    });
+};
+
+/**
+* function that actually broadcasts hub messages to websocket
+* @async
+*/
+async function receiveHubMessages() 
+  {
+      const consumerGroup = "$Default"; // name of the default consumer group
+      eventHubReader = new EventHubReader(IOTHUB_CONNECTION_STRING, consumerGroup);
+      (async () => 
+      {
+          await eventHubReader.startReadMessage((message, date, deviceId) => 
+          {
+            try 
+            {
+              const payload = 
+              {
+                IotData: message,
+                MessageDate: date || Date.now().toISOString(),
+                DeviceId: deviceId,
+              };
+
+              wss.broadcast(JSON.stringify(payload));
+            } catch (error) 
+            {
+              console.error('Error broadcasting: [%s] from [%s].', error, message);
+            }
+
+          });
+      }).then((response) =>
+      {
+        //do nothing, already handled
+      }).catch((error) =>
+        {
+          console.error('Error in receiveHubMessages: ', error);
+        });
+}
+
+/**
+ * Connects to IoT Hub and returns a list of currently running modules 
+ * @returns {Promise<ResultWithHttpResponse<Module[]>>}
+ */
+function iotHubConnection(req, res)
```

Comment:
req/res aren't used. also this method is only used in one spot and is short so might not want it at all.

Reply:
looks like there's a few places the params you're sending aren't used, check more broadly as well

Reply:
resolved